writer.go

```diff
@@ -17,6 +18,9 @@ type Writer struct {
 	framer    Framer
 	formatter Formatter
 
+	//non-nil if custom dialer set, used in getDialer
+	cdialer func(string, string) (net.Conn, error)
```

Comment:
Is `cdialer` the ideal name for this? Or would `customDialer` be more explanatory?

Reply:
I thought it was self-explanatory, but I don't mind the longer name either.

Reply:
Maybe it's just my eyes/brain being dumb, but every time I read it, it sounds like "C Dialer" (as in somehow related to C), rather than "custom dialer".